model/shared/src/main/scala/explore/CallForProposal.scala

```diff
@@ -36,13 +37,24 @@ case class CallForProposal(
 ) derives Eq,
       Decoder:
 
-  def deadline(proposalPartners: List[Partner]): Option[Timestamp] =
-    val callPartners = partners
-    proposalPartners
-      .map(p => callPartners.find(_.partner === p).flatMap(_.submissionDeadline))
-      .minimumOption
-      .flatten
-      .orElse(submissionDeadlineDefault)
+  def deadline(piPartner: Option[PartnerLink]): Option[Timestamp] =
+    piPartner.flatMap {
+      _.fold(
+        None,
+        partners
+          .find(_.partner === Partner.US)
+          .flatMap(_.submissionDeadline)
+          .orElse(submissionDeadlineDefault),
```

Comment:
The call for proposals has a `nonPartnerDeadline` field.  Also the partner `submissionDeadline` already does the explicit deadline or else default in the view from which the data comes.